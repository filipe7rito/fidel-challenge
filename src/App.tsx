import React from 'react';
import { css } from 'emotion';

function App() {
  const thing = 'world';

  return <div className={style}>{`Hello ${thing}`}</div>;
}

export default App;

const style = css`
  color: royalblue;
`;
