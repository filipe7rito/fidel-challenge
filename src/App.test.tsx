import { render } from '@testing-library/react';
import React from 'react';
import App from './App';

describe('App', () => {
  test('Render app', () => {
    const { asFragment } = renderPersons();

    expect(asFragment()).toMatchSnapshot();
  });

  function renderPersons() {
    return render(<App />);
  }
});
