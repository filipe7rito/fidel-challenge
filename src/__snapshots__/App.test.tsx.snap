// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Render app 1`] = `
<DocumentFragment>
  <div
    class="css-1rpfbq7"
  >
    <div
      class="css-1d6fn9t"
    >
      <main>
        <div
          class="css-gaa0i9"
        >
          <h2>
            Transactions
          </h2>
          <div
            class="css-1affd20"
          >
            <span
              class="ant-input-affix-wrapper css-1joxlfe"
            >
              <span
                class="ant-input-prefix"
              >
                <span
                  aria-label="search"
                  class="anticon anticon-search"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="search"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                    />
                  </svg>
                </span>
              </span>
              <input
                class="ant-input"
                placeholder="Filter table results"
                type="text"
                value=""
              />
              <span
                class="ant-input-suffix"
              >
                <span
                  aria-label="close-circle"
                  class="anticon anticon-close-circle ant-input-clear-icon-hidden ant-input-clear-icon"
                  role="button"
                  tabindex="-1"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="close-circle"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                    />
                  </svg>
                </span>
              </span>
            </span>
            <div
              class="ant-spin-nested-loading"
            >
              <div>
                <div
                  class="ant-spin ant-spin-spinning ant-spin-show-text"
                >
                  <span
                    class="ant-spin-dot ant-spin-dot-spin"
                  >
                    <i
                      class="ant-spin-dot-item"
                    />
                    <i
                      class="ant-spin-dot-item"
                    />
                    <i
                      class="ant-spin-dot-item"
                    />
                    <i
                      class="ant-spin-dot-item"
                    />
                  </span>
                  <div
                    class="ant-spin-text"
                  >
                    Loading...
                  </div>
                </div>
              </div>
              <div
                class="ant-spin-container ant-spin-blur"
              >
                <table>
                  <thead>
                    <tr
                      role="row"
                    >
                      <th
                        class="css-mxi0c8"
                        colspan="1"
                        role="columnheader"
                      >
                        <div>
                          Amount
                        </div>
                      </th>
                      <th
                        class="css-mxi0c8"
                        colspan="1"
                        role="columnheader"
                      >
                        <div>
                          Address
                        </div>
                      </th>
                      <th
                        class="css-mxi0c8"
                        colspan="1"
                        role="columnheader"
                      >
                        <div>
                          Scheme
                        </div>
                      </th>
                      <th
                        class="css-mxi0c8"
                        colspan="1"
                        role="columnheader"
                      >
                        <div>
                          Card
                        </div>
                      </th>
                      <th
                        class="css-mxi0c8"
                        colspan="1"
                        role="columnheader"
                      >
                        <div>
                          Date(UTC)
                        </div>
                      </th>
                      <th
                        class="css-mxi0c8"
                        colspan="1"
                        role="columnheader"
                      >
                        <div>
                          <span
                            aria-label="reload"
                            class="anticon anticon-reload css-1tya5c2"
                            role="img"
                            tabindex="-1"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="reload"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M909.1 209.3l-56.4 44.1C775.8 155.1 656.2 92 521.9 92 290 92 102.3 279.5 102 511.5 101.7 743.7 289.8 932 521.9 932c181.3 0 335.8-115 394.6-276.1 1.5-4.2-.7-8.9-4.9-10.3l-56.7-19.5a8 8 0 00-10.1 4.8c-1.8 5-3.8 10-5.9 14.9-17.3 41-42.1 77.8-73.7 109.4A344.77 344.77 0 01655.9 829c-42.3 17.9-87.4 27-133.8 27-46.5 0-91.5-9.1-133.8-27A341.5 341.5 0 01279 755.2a342.16 342.16 0 01-73.7-109.4c-17.9-42.4-27-87.4-27-133.9s9.1-91.5 27-133.9c17.3-41 42.1-77.8 73.7-109.4 31.6-31.6 68.4-56.4 109.3-73.8 42.3-17.9 87.4-27 133.8-27 46.5 0 91.5 9.1 133.8 27a341.5 341.5 0 01109.3 73.8c9.9 9.9 19.2 20.4 27.8 31.4l-60.2 47a8 8 0 003 14.1l175.6 43c5 1.2 9.9-2.6 9.9-7.7l.8-180.9c-.1-6.6-7.8-10.3-13-6.2z"
                              />
                            </svg>
                          </span>
                        </div>
                      </th>
                    </tr>
                  </thead>
                  <tbody />
                </table>
                <div
                  class="css-a0z5ei"
                >
                  Showing 0 of 0
                </div>
              </div>
            </div>
          </div>
          <div
            class="css-1viymee"
          >
            <div
              class=""
            >
              <div
                class="ant-drawer ant-drawer-right"
                style="position: absolute;"
                tabindex="-1"
              >
                <div
                  class="ant-drawer-mask"
                />
                <div
                  class="ant-drawer-content-wrapper"
                  style="transform: translateX(100%); width: 570px;"
                >
                  <div
                    class="ant-drawer-content"
                  >
                    <div
                      class="ant-drawer-wrapper-body"
                    >
                      <div
                        class="ant-drawer-header"
                      >
                        <div
                          class="ant-drawer-title"
                        >
                          Transaction details
                        </div>
                        <button
                          aria-label="Close"
                          class="ant-drawer-close"
                          style="--scroll-bar: 0px;"
                          type="button"
                        >
                          <span
                            aria-label="close"
                            class="anticon anticon-close"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="close"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                              />
                            </svg>
                          </span>
                        </button>
                      </div>
                      <div
                        class="ant-drawer-body"
                      >
                        <div
                          class="css-ppcisl"
                        >
                          <div
                            class="ant-spin ant-spin-spinning"
                          >
                            <span
                              class="ant-spin-dot ant-spin-dot-spin"
                            >
                              <i
                                class="ant-spin-dot-item"
                              />
                              <i
                                class="ant-spin-dot-item"
                              />
                              <i
                                class="ant-spin-dot-item"
                              />
                              <i
                                class="ant-spin-dot-item"
                              />
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
    <footer
      class="footer"
    />
  </div>
</DocumentFragment>
`;
